frame/benchmark/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// A runtime module template with necessary imports
+
+/// Feel free to remove or edit this file as needed.
+/// If you change the name of this file, make sure to update its references in runtime/src/lib.rs
+/// If you remove this file, you can remove those references
+
+/// For more guidance on Substrate modules, see the example module
+/// https://github.com/paritytech/substrate/blob/master/frame/example/src/lib.rs
+
+use frame_support::{decl_module, decl_storage, decl_event, decl_error};
+use frame_support::traits::Currency;
+use frame_system::{self as system, ensure_signed};
+use codec::{Encode, Decode};
+use sp_std::prelude::Vec;
+
+pub mod benchmarking;
+
+/// Type alias for currency balance.
+pub type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+
+/// The pallet's configuration trait.
+pub trait Trait: system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	type Currency: Currency<Self::AccountId>;
+}
+
+// This pallet's storage items.
+decl_storage! {
+	trait Store for Module<T: Trait> as Benchmark {
+		MyMemberList: Vec<T::AccountId>;
+		MyMemberMap: map hasher(blake2_256) T::AccountId => bool;
+		MyValue: u32;
+		MyMap: map hasher(blake2_256) u32 => u32;
+		MyDoubleMap: double_map hasher(blake2_256) u32, hasher(blake2_256) u32 => u32;
+	}
+}
+
+// The pallet's events
+decl_event!(
+	pub enum Event<T> where AccountId = <T as system::Trait>::AccountId {
+		Dummy(u32, AccountId),
+	}
+);
+
+// The pallet's errors
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// Value was None
+		NoneValue,
+		/// Value reached maximum and cannot be incremented further
+		StorageOverflow,
+	}
+}
+
+// The pallet's dispatchable functions.
+decl_module! {
+	/// The module declaration.
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		// Initializing errors
+		// this includes information about your errors in the node's metadata.
+		// it is needed only if you are using errors in your pallet
```

Comment:
Some of these docs are probably pulled in unintentionally from https://github.com/paritytech/substrate/blob/5a2824d9b20f151466d92f20c767141c30a35b65/bin/node-template/pallets/template/src/lib.rs#L65 and such